llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -486,10 +492,19 @@ def dtensor_to_tensor_hook(
             )
 
             log.debug('Saving Hugging Face checkpoint to disk')
-            new_model_instance.save_pretrained(temp_save_dir)
-            if original_tokenizer is not None:
-                assert isinstance(original_tokenizer, PreTrainedTokenizerBase)
-                original_tokenizer.save_pretrained(temp_save_dir)
+            # This context manager casts the TE extra state in io.BytesIO format to tensor format
+            # Needed for proper hf ckpt saving.
+            context_manager = te.onnx_export(
```

Comment:
is there a more granular check we can do to decide whether to wrap in onnx export context manager? I'd prefer to only wrap if the model _actually_ needs it. not just if te is importable.

Reply:
added precision state check